tests/test_directives.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) 2017 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import pytest
+import sys
+from copy import deepcopy
+
+from flow.directives import Directives, _DirectivesItem
+from flow.directives import NP, NRANKS, NGPU, EXECUTABLE, OMP_NUM_THREADS
+from flow.directives import WALLTIME, MEMORY, PROCESS_FRACTION
+
+
+@pytest.fixture()
+def setUp(test_class=Directives, **kwargs):
+    def _setup(test_class=Directives, **kwargs):
+        return test_class(**kwargs)
+    return _setup
+
+@pytest.fixture()
+def available_directives_list():
+    return [NP, NRANKS, NGPU, OMP_NUM_THREADS,
+            EXECUTABLE, WALLTIME, MEMORY,
+            PROCESS_FRACTION]
+
+
+class TestItems:
+    """
+    Tests for _DirectivesItem class
+    """
+    def test_default(self):
+        assert NP.default == 1
+        assert NGPU.default == 0
+        assert NRANKS.default == 0
+        assert OMP_NUM_THREADS.default == 0
+        assert EXECUTABLE.default == sys.executable
+        assert WALLTIME.default == 12.0
+        assert MEMORY.default == 4
+        assert PROCESS_FRACTION.default == 1.
+
+    def test_validation(self, available_directives_list):
+        for directive in available_directives_list:
+            directive.validation(directive.default)
+
+    # def test_validation_none(self):
+    #     available_directives_list =
+    #     for directive in available_directives_list:
+    #         directive.validation(directive.default)
+
+    def test_validation_manual(self):
+        directive_copy = deepcopy(NP)
+
+        def div_by_ten(v):
```

Comment:
technically this is mod not divide.

Reply:
Also, this does test the functionality; however, the conversion from a number to whether it is divisible by ten may be confusing since the purpose is to return a validated value or raise an error.

Reply:
Resolved it locally, by returning `v` itself if we is divisible by 10 else we raise an error.